datastream-kafka-connector/src/test/java/com/linkedin/datastream/connectors/kafka/mirrormaker/TestKafkaTopicPartitionStats.java

```diff
@@ -0,0 +1,148 @@
+import com.linkedin.datastream.connectors.kafka.MockDatastreamEventProducer;
+import com.linkedin.datastream.server.DatastreamTaskImpl;
+import com.linkedin.datastream.testutil.BaseKafkaZkTest;
+
+
+/**
+ * Tests for TopicPartitionStats diag command
+ */
+@Test
+public class TestKafkaTopicPartitionStats extends BaseKafkaZkTest {
+
+  private static final int PARTITION_COUNT = 2;
+  private static final String TOPICPARTITION_STATS_QUERY = "/topicpartition_stats";
+
+  @Test
+  public void testProcessTopicPartitionStats() {
+    List<String> topics = Arrays.asList("topic1", "topic2");
+    topics.forEach(topic -> createTopic(_zkUtils, topic, PARTITION_COUNT));
+
+    Datastream datastream = KafkaMirrorMakerConnectorTestUtils.createDatastream("topicStream",  _broker, "topic\\d+");
+
+    DatastreamTaskImpl task = new DatastreamTaskImpl(Collections.singletonList(datastream));
+    MockDatastreamEventProducer datastreamProducer = new MockDatastreamEventProducer();
+    task.setEventProducer(datastreamProducer);
+
+    KafkaMirrorMakerConnector connector =
+        new KafkaMirrorMakerConnector("MirrorMakerConnector",
+            KafkaMirrorMakerConnectorTestUtils.getDefaultConfig(Optional.empty()), "testCluster");
+    connector.start(null);
+
+    // Notify connector of paused partition update
+    connector.onAssignmentChange(Collections.singletonList(task));
+
+    Set<TopicPartition> expected = new HashSet<>();
+    topics.forEach(topic -> {
+      for (int i = 0; i < PARTITION_COUNT; ++i) {
+        expected.add(new TopicPartition(topic, i));
+      }
+    });
+
+    // Wait until the partitions are assigned
+    if (!PollUtils.poll(() -> testProcessTopicPartitionsStatsInternal(connector, expected),
+        KafkaMirrorMakerConnectorTestUtils.POLL_PERIOD_MS,
+        KafkaMirrorMakerConnectorTestUtils.POLL_TIMEOUT_MS)) {
+      Assert.fail("Topic partitions still not assigned");
+    }
+
+    // Delete the topic to revoke partitions from topic2
+    deleteTopic(_zkUtils, topics.get(0));
+    expected.clear();
+    for (int i = 0; i < PARTITION_COUNT; ++i) {
+      expected.add(new TopicPartition(topics.get(1), i));
+    }
+
+    // Wait until the partitions from the deleted topic are revoked
+    if (!PollUtils.poll(() -> testProcessTopicPartitionsStatsInternal(connector, expected),
+        KafkaMirrorMakerConnectorTestUtils.POLL_PERIOD_MS,
+        KafkaMirrorMakerConnectorTestUtils.POLL_TIMEOUT_MS)) {
+      Assert.fail("Topic partitions still not revoked");
+    }
+    connector.stop();
+  }
+
+  boolean testProcessTopicPartitionsStatsInternal(KafkaMirrorMakerConnector connector, Set<TopicPartition> expected) {
+    String jsonStr = connector.process("/topicpartition_stats");
+    List<KafkaTopicPartitionStatsResponse> responseList = KafkaTopicPartitionStatsResponse.fromJson(jsonStr);
+
+    Set<TopicPartition> actual = new HashSet<>();
+    responseList.forEach(response -> {
+      actual.addAll(response.getTopicPartitions());
+    });
```

Comment:
nit: You don't need parenthesis for one liner lambdas
```java
    responseList.forEach(response -> actual.addAll(response.getTopicPartitions()));
```

Reply:
Thanks! I will retain this as it makes it more clearer.